packages/api/src/controllers/access-control.ts

```diff
@@ -50,73 +57,184 @@ app.post(
+            );
+        }
+      default:
         console.log(`
-          access-control: gate: content contentId ${content.id} with playbackId=${playbackId} is gated but multiple (${signingKeyOutput.length}) public keys found for key ${req.body.pub}, disallowing playback, colliding keys=${collisionKeys}
+          access-control: gate: content with playbackId ${playbackId} is gated but playbackPolicyType ${playbackPolicyType} is not supported, disallowing playback
         `);
         throw new BadRequestError(
-          "Multiple signing keys found for the same public key."
+          `unknown playbackPolicy type: ${playbackPolicyType}`
         );
-      }
-
-      const signingKey = signingKeyOutput[0];
+    }
+  }
+);
 
-      if (signingKey.userId !== content.userId) {
-        console.log(`
-          access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
-        `);
-        throw new NotFoundError("Content not found");
-      }
+async function fireGateWebhook(
+  webhook: DBWebhook,
+  plabackPolicy: PlaybackPolicy,
+  accessKey: string
+) {
+  let timestamp = Date.now();
+  let params = {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+      "user-agent": "livepeer.studio",
+    },
+    timeout: WEBHOOK_TIMEOUT,
+    body: JSON.stringify({
+      context: plabackPolicy.webhookContext,
+      accessKey: accessKey,
+    }),
+  };
+
+  if (webhook.sharedSecret) {
+    params = await addSignatureHeader(params, webhook, timestamp);
+  }
 
-      if (signingKey.disabled || signingKey.deleted) {
-        const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
-        console.log(`
-          access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
-        `);
-        throw new ForbiddenError("The public key is disabled or deleted");
-      }
-
-      tracking.recordSigningKeyValidation(signingKey.id);
-      res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
-      res.status(204);
-      return res.end();
-    } else {
-      console.log(`
-        access-control: gate: content with playbackId ${playbackId} is gated but playbackPolicyType ${playbackPolicyType} is not supported, disallowing playback
-      `);
-      throw new BadRequestError(
-        `unknown playbackPolicy type: ${playbackPolicyType}`
-      );
+  const startTime = process.hrtime();
+  let resp: Response;
+  let errorMessage: string;
+  let responseBody: string;
+  let statusCode: number;
+
+  try {
+    console.log(`webhook ${webhook.id} firing`);
+    resp = await fetchWithTimeout(webhook.url, params);
+    responseBody = await resp.text();
```

Comment:
We should not read or store this, one can probably exhaust our DB by returning large bodies here. The response payload should be completely ignored, we should only check the status code.

Also check if we need to "close" the response in some way if we ignore the body. Not sure how that disposal works in node 🤔